genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go

```diff
@@ -0,0 +1,92 @@
+package idp_salesforce
+
+import (
+	"context"
+	"github.com/mypurecloud/platform-client-sdk-go/v129/platformclientv2"
+)
+
+/*
+The genesyscloud_idp_salesforce_proxy.go file contains the proxy structures and methods that interact
+with the Genesys Cloud SDK. We use composition here for each function on the proxy so individual functions can be stubbed
+out during testing.
+*/
+
+// internalProxy holds a proxy instance that can be used throughout the package
+var internalProxy *idpSalesforceProxy
+
+// Type definitions for each func on our proxy so we can easily mock them out later
+type getIdpSalesforceByIdFunc func(ctx context.Context, p *idpSalesforceProxy) (salesforce *platformclientv2.Salesforce, resp *platformclientv2.APIResponse, err error)
+type updateIdpSalesforceFunc func(ctx context.Context, p *idpSalesforceProxy, salesforce *platformclientv2.Salesforce) (*platformclientv2.Identityprovider, *platformclientv2.APIResponse, error)
+type deleteIdpSalesforceFunc func(ctx context.Context, p *idpSalesforceProxy) (resp *platformclientv2.APIResponse, err error)
+
+// idpSalesforceProxy contains all of the methods that call genesys cloud APIs.
+type idpSalesforceProxy struct {
+	clientConfig             *platformclientv2.Configuration
+	identityProviderApi      *platformclientv2.IdentityProviderApi
+	getIdpSalesforceByIdAttr getIdpSalesforceByIdFunc
+	updateIdpSalesforceAttr  updateIdpSalesforceFunc
+	deleteIdpSalesforceAttr  deleteIdpSalesforceFunc
+}
+
+// newIdpSalesforceProxy initializes the idp salesforce proxy with all of the data needed to communicate with Genesys Cloud
+func newIdpSalesforceProxy(clientConfig *platformclientv2.Configuration) *idpSalesforceProxy {
+	api := platformclientv2.NewIdentityProviderApiWithConfig(clientConfig)
+	return &idpSalesforceProxy{
+		clientConfig:             clientConfig,
+		identityProviderApi:      api,
+		getIdpSalesforceByIdAttr: getIdpSalesforceByIdFn,
+		updateIdpSalesforceAttr:  updateIdpSalesforceFn,
+		deleteIdpSalesforceAttr:  deleteIdpSalesforceFn,
+	}
+}
+
+// getIdpSalesforceProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getIdpSalesforceProxy(clientConfig *platformclientv2.Configuration) *idpSalesforceProxy {
+	if internalProxy == nil {
+		internalProxy = newIdpSalesforceProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getIdpSalesforceById returns a single Genesys Cloud idp salesforce by Id
+func (p *idpSalesforceProxy) getIdpSalesforceById(ctx context.Context) (idpSalesforce *platformclientv2.Salesforce, resp *platformclientv2.APIResponse, err error) {
```

Comment:
"ById" isn't really appropriate here. `getIdpSalesforce` makes more sense

Reply:
Thanks Charlie, I forgot to change that